File: aula7/python_logic/desafio_final_mvp.py
def exibir_cabecalho():
    # Essa função centraliza o cabeçalho para evitar repetição visual no sistema
    print("\n===================================")
    print("      SORVETERIA DO DENER")
    print("===================================")

def verificar_estoque_critico(quantidade):
    # Aqui a função recebe a quantidade e devolve uma mensagem de auditoria
    if quantidade < 5:
        return " [REPOSIÇÃO NECESSÁRIA]"
    return ""

def listar_produtos(nomes, quantidades):
    # Essa função percorre as listas em paralelo para montar o relatório
    print("\nRelatório de estoque:")
    for i in range(len(nomes)):
        alerta = verificar_estoque_critico(quantidades[i])
        print(f"Produto: {nomes[i]} | Estoque: {quantidades[i]} unidades{alerta}")

nomes = []
quantidades = []

exibir_cabecalho()

# Primeiro cadastro inicial de 3 insumos, seguindo a ideia do MVP das aulas anteriores
for i in range(3):
    nome = input(f"Digite o nome do {i + 1}º produto: ")
    quantidade = int(input(f"Digite a quantidade de {nome}: "))
    nomes.append(nome)
    quantidades.append(quantidade)

opcao = 0

# Agora o while mantém o sistema ativo para processar a lista de insumos
while opcao != 3:
    print("\n1. Listar estoque")
    print("2. Adicionar novo produto")
    print("3. Sair")
    opcao = int(input("Escolha uma opção: "))

    if opcao == 1:
        exibir_cabecalho()
        listar_produtos(nomes, quantidades)
    elif opcao == 2:
        nome = input("Digite o nome do novo produto: ")
        quantidade = int(input(f"Digite a quantidade de {nome}: "))
        nomes.append(nome)
        quantidades.append(quantidade)
        print("Produto cadastrado com sucesso.")
    elif opcao == 3:
        print("Sistema encerrado.")
    else:
        print("Opção inválida.")

File: aula7/cpp_logic/desafio_final_mvp.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

void exibirCabecalho() {
    // Essa função cuida só da identidade visual do terminal
    cout << "\n===================================" << endl;
    cout << "      SORVETERIA DO DENER" << endl;
    cout << "===================================" << endl;
}

string verificarEstoqueCritico(int quantidade) {
    // Aqui a função retorna a etiqueta de reposição quando necessário
    if (quantidade < 5) {
        return " [REPOSICAO NECESSARIA]";
    }
    return "";
}

void listarProdutos(vector<string>& nomes, vector<int>& quantidades) {
    // Estou percorrendo os dois vetores em paralelo para montar o relatório
    cout << "\nRelatorio de estoque:" << endl;
    for (int i = 0; i < nomes.size(); i++) {
        string alerta = verificarEstoqueCritico(quantidades[i]);
        cout << "Produto: " << nomes[i] << " | Estoque: " << quantidades[i] << " unidades" << alerta << endl;
    }
}

int main() {
    vector<string> nomes;
    vector<int> quantidades;

    exibirCabecalho();

    for (int i = 0; i < 3; i++) {
        string nome;
        int quantidade;

        cout << "Digite o nome do " << (i + 1) << " produto: ";
        getline(cin, nome);

        cout << "Digite a quantidade de " << nome << ": ";
        cin >> quantidade;
        cin.ignore();

        nomes.push_back(nome);
        quantidades.push_back(quantidade);
    }

    int opcao = 0;

    while (opcao != 3) {
        cout << "\n1. Listar estoque" << endl;
        cout << "2. Adicionar novo produto" << endl;
        cout << "3. Sair" << endl;
        cout << "Escolha uma opcao: ";
        cin >> opcao;
        cin.ignore();

        if (opcao == 1) {
            exibirCabecalho();
            listarProdutos(nomes, quantidades);
        } else if (opcao == 2) {
            string nome;
            int quantidade;

            cout << "Digite o nome do novo produto: ";
            getline(cin, nome);

            cout << "Digite a quantidade de " << nome << ": ";
            cin >> quantidade;
            cin.ignore();

            nomes.push_back(nome);
            quantidades.push_back(quantidade);

            cout << "Produto cadastrado com sucesso." << endl;
        } else if (opcao == 3) {
            cout << "Sistema encerrado." << endl;
        } else {
            cout << "Opcao invalida." << endl;
        }
    }

    return 0;
}